idaes/models/unit_models/tests/test_heat_exchanger_1D.py

```diff
@@ -703,44 +703,57 @@ def test_conservation(self, btx):
 
 
 # -----------------------------------------------------------------------------
-@pytest.mark.iapws
-@pytest.mark.skipif(not iapws95.iapws95_available(), reason="IAPWS not available")
-class TestIAPWS_cocurrent(object):
-    @pytest.fixture(scope="class")
-    def iapws(self):
-        m = ConcreteModel()
-        m.fs = FlowsheetBlock(default={"dynamic": False})
+def build_model():
+    m = ConcreteModel()
+    m.fs = FlowsheetBlock(default={"dynamic": False})
 
-        m.fs.properties = iapws95.Iapws95ParameterBlock(
-            default={"phase_presentation": iapws95.PhaseType.LG}
-        )
+    m.fs.properties = iapws95.Iapws95ParameterBlock(
+        default={"phase_presentation": iapws95.PhaseType.LG}
+    )
 
-        m.fs.unit = HX1D(
-            default={
-                "shell_side": {"property_package": m.fs.properties},
-                "tube_side": {"property_package": m.fs.properties},
-                "flow_type": HeatExchangerFlowPattern.cocurrent,
-            }
-        )
+    m.fs.unit = HX1D(
+        default={
+            "shell_side": {"property_package": m.fs.properties},
+            "tube_side": {"property_package": m.fs.properties},
+            "flow_type": HeatExchangerFlowPattern.cocurrent,
+        }
+    )
 
-        m.fs.unit.d_shell.fix(1.04)
-        m.fs.unit.d_tube_outer.fix(0.01167)
-        m.fs.unit.d_tube_inner.fix(0.01067)
-        m.fs.unit.N_tubes.fix(10)
-        m.fs.unit.shell_length.fix(4.85)
-        m.fs.unit.tube_length.fix(4.85)
-        m.fs.unit.shell_heat_transfer_coefficient.fix(2000)
-        m.fs.unit.tube_heat_transfer_coefficient.fix(51000)
+    m.fs.unit.d_shell.fix(1.04)
+    m.fs.unit.d_tube_outer.fix(0.01167)
+    m.fs.unit.d_tube_inner.fix(0.01067)
+    m.fs.unit.N_tubes.fix(10)
+    m.fs.unit.shell_length.fix(4.85)
+    m.fs.unit.tube_length.fix(4.85)
+    m.fs.unit.shell_heat_transfer_coefficient.fix(2000)
+    m.fs.unit.tube_heat_transfer_coefficient.fix(51000)
 
-        m.fs.unit.shell_inlet.flow_mol[0].fix(5)
-        m.fs.unit.shell_inlet.enth_mol[0].fix(50000)
-        m.fs.unit.shell_inlet.pressure[0].fix(101325)
+    m.fs.unit.shell_inlet.flow_mol[0].fix(5)
+    m.fs.unit.shell_inlet.enth_mol[0].fix(50000)
+    m.fs.unit.shell_inlet.pressure[0].fix(101325)
 
-        m.fs.unit.tube_inlet.flow_mol[0].fix(5)
-        m.fs.unit.tube_inlet.enth_mol[0].fix(7000)
-        m.fs.unit.tube_inlet.pressure[0].fix(101325)
+    m.fs.unit.tube_inlet.flow_mol[0].fix(5)
+    m.fs.unit.tube_inlet.enth_mol[0].fix(7000)
+    m.fs.unit.tube_inlet.pressure[0].fix(101325)
 
-        return m
+    return m
+
+
+@pytest.mark.performance
+class Test_HX1D_Performance(PerformanceBaseClass):
```

Comment:
Can these test classes also inherit from `unittest.TestCase`, e.g.
```python
class Test_HX1D_Performance(PerformanceBaseClass, unittest.TestCase):
    ...
```
This way they are discoverable with `unittest.TestLoader.discover`. The reason to put the inheritance on the classes for specific models is so that the "discovery script" I posted above generates only these classes, rather than `PerformanceBaseClass` as well. This is inelegant and could be avoided if we could perform this discovery without relying on unittest. Presumably this is possible with pytest, probably via hooks/plugins (`pytest_collection`?) or the `pytest.Collector` class, but I wasn't able to get it working in a couple hours of trying.

Reply:
Yes, that should be possible. I had intended to do this after the change in the comment below.